tracer/src/Datadog.Trace/PDBs/DatadogMetadataReader.cs

```diff
@@ -366,14 +366,26 @@ private MethodDefinitionHandle GetMoveNextMethod(MethodDefinition methodDef)
 
             if (PdbReader != null)
             {
+                var normalizeFilePath = GetNormalizedPath(filePath);
+                if (string.IsNullOrEmpty(normalizeFilePath))
+                {
+                    return null;
+                }
+
                 const int methodDefTablePrefix = 0x06000000;
                 foreach (MethodDefinitionHandle methodDefinitionHandle in MetadataReader.MethodDefinitions)
                 {
                     MethodDebugInformation methodDebugInformation = PdbReader.GetMethodDebugInformation(methodDefinitionHandle);
 
                     foreach (VendoredMicrosoftCode.System.Reflection.Metadata.SequencePoint sequencePoint in methodDebugInformation.GetSequencePoints())
                     {
-                        if (sequencePoint.IsHidden || GetDocumentName(sequencePoint.Document) != filePath)
+                        var normalizeDocumentPath = GetNormalizedPath(GetDocumentName(sequencePoint.Document));
+                        if (!string.Equals(normalizeDocumentPath, normalizeFilePath, StringComparison.OrdinalIgnoreCase))
+                        {
+                            break;
```

Comment:
1. Because it's per method so if the first not-hidden sp is not belong to the require path no need to check other sequnce points of this method
2. It go over all methods in the current metadata reader..
3. I've moved the hidden check before the path check to be sure not skip in case of hidden sp does not has path
 9f23fdedaa2a7645d2ae2d2edbaf79936c9e57ef